packages/core/src/components/Layout.tsx

```diff
@@ -104,7 +105,9 @@ export default function Layout({
     })
   }
 
-  const [isMobileView] = useMediaQuery('(max-width: 480px)')
+  useEffect(() => {
```

Comment:
Hey @betty-alagwu ! What does this useEffect do? If it's just for handling dimensions change on a device, I think we don't need it. Not a lot of users changing their device's dimensions, except for us testers.

Would just setting the initial value correctly on Line 37 work?

Something like this 

```
const [isSidebarCollapsed, setIsSidebarCollapsed] = useState(false || isMobileView)
```

I wasn't also able to test this out completely. But I think with this useEffect, it will also prevent the sidebar from opening up again on desktop devices once it has been closed - because it'll be triggered every time `setIsSidebarCollapsed` variable is changed, and set it to false.

Reply:
Hello @Pranav2612000 ! Thanks for flagging this. 

The `useEffect` hook helps keep the sidebar's collapsed state in sync with the device's mobile view status.

Setting the initial value directly will not produce the desired behaviour and this is because the initial value of `isMobileView` might not be available immediately since the `useMediaQuery` hook takes some time to determine if the device is in mobile view or not.

Also while the `useEffect` is updating `setIsSidebarCollapsed`, it does not prevent the sidebar from opening up on desktop, such as when the `isMobileView` changes, the sidebar will reopen on desktop devices.

Here's a short video of how it works now:


https://github.com/Mirrorful/mirrorful/assets/94234459/f27f4627-bc1b-4387-8685-6040fcd26d46